radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java

```diff
@@ -825,6 +768,62 @@ public UnsignedAtom buildAtomWithFee(List<ParticleGroup> particleGroups) {
 		return new UnsignedAtom(new Atom(allParticleGroups, metaData));
 	}
 
+	/**
+	 * Represents an atomic transaction to be committed to the ledger
+	 */
+	public final class Transaction {
+		private final ArrayList<Action> actions = new ArrayList<>();
+		private Transaction() {
+		}
+
+		/**
+		 * Execute an action within this transaction
+		 * @param action the action to execute
+		 */
+		public void execute(Action action) {
```

Comment:
I have just reviewed the regression tests where I got slightly confused by the wording "execute". I first thought it really meant "execute" (as in "do something - now"). I then remembered that "execute" in this case means "add" or "enqueue" or "stash" (no, not stash) or "save-for-later". Maybe we could rename this method to something that better reflects what's actually going on. `addAction` could be a classical alternative.

Reply:
Also could this return `this` to allow a builder (-ish) pattern?

```java
public Completable prepareTransaction() {
    return api.transaction()
            .execute(new SomeAction())
            .execute(new SomeOtherAction())
            .commit()
            .toCompletable();
}
```

Reply:
Fair point!